dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/PartitionUtils.java

```diff
@@ -251,11 +260,26 @@ public static AnalyticsTablePartition getLatestTablePartition( List<AnalyticsTab
      * Returns partition name. Aggregate only for now!
      *
      * @param tableName the table name.
-     * @param partitiont the partition.
+     * @param partition the partition.
      * @return the partition name.
      */
     public static String getPartitionName( String tableName, Integer partition )
     {
         return tableName + SEP + partition;
     }
+
+    /**
+     * Returns the year of the first week of this period
+     *
+     * @param period a {@see Period}
+     *
+     * @return the 4 digits representation of an year
+     */
+    private static int getYearOfFirstWeekFromPeriod( Period period )
```

Comment:
Are we taking the various weekly period types (starting Tuesday, Wednesday, etc) here?